src/Service/InvoicingRateReportService.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+namespace App\Service;
+
+use App\Model\Reports\InvoicingRateReportData;
+use App\Model\Reports\InvoicingRateReportViewModeEnum;
+use App\Model\Reports\InvoicingRateReportWorker;
+use App\Model\Reports\WorkloadReportPeriodTypeEnum as PeriodTypeEnum;
+use App\Repository\WorkerRepository;
+use App\Repository\WorklogRepository;
+
+class InvoicingRateReportService
+{
+    public function __construct(
+        private readonly WorkerRepository $workerRepository,
+        private readonly WorklogRepository $worklogRepository,
+        private readonly DateTimeHelper $dateTimeHelper,
+    ) {
+    }
+
+    /**
+     * Generates an invoicing rate report for a specific year based on various parameters.
+     *
+     * @param int $year the year for which the report is generated
+     * @param PeriodTypeEnum $viewPeriodType the period type
+     * @param InvoicingRateReportViewModeEnum $viewMode the view mode
+     * @param bool $includeIssues whether to include detailed issue-level data in the report
+     *
+     * @return InvoicingRateReportData the calculated invoicing rate report data
+     *
+     * @throws \Exception if a required worker identifier is empty
+     */
+    public function getInvoicingRateReport(
+        int $year,
+        PeriodTypeEnum $viewPeriodType = PeriodTypeEnum::WEEK,
+        InvoicingRateReportViewModeEnum $viewMode = InvoicingRateReportViewModeEnum::SUMMARY,
+        bool $includeIssues = false,
+    ): InvoicingRateReportData {
+        $invoicingRateReportData = new InvoicingRateReportData($viewPeriodType->value);
+        $invoicingRateReportData->includeIssues = $includeIssues;
+        if (!$year) {
+            $year = (int) (new \DateTime())->format('Y');
+        }
+        $workers = $this->workerRepository->findAllIncludedInReports();
+        $periods = $this->getPeriods($viewPeriodType, $year);
+        $periodSums = [];
+        $periodCounts = [];
+
+        foreach ($periods as $period) {
+            $readablePeriod = $this->getReadablePeriod($period, $viewPeriodType);
+            $invoicingRateReportData->period->set((string) $period, $readablePeriod);
+        }
+
+        foreach ($workers as $worker) {
+            $invoicingRateReportWorker = new InvoicingRateReportWorker();
+            $invoicingRateReportWorker->setEmail($worker->getUserIdentifier());
+            $invoicingRateReportWorker->setWorkload($worker->getWorkload());
+            $invoicingRateReportWorker->setName($worker->getName());
+            $currentPeriodReached = false;
+            $loggedBilledHoursSum = 0;
+            $loggedHoursSum = 0;
+            $workerProjects = [];
+
+            foreach ($periods as $period) {
+                // Add current period match-point (current week-number, month-number etc.)
```

Comment:
Please add some more explanation about why it should be 53 when not the current year.